### PR TITLE
Specify behavior of default this initializer with field initializers

### PR DESCRIPTION
**Pull request by RikkiGibson**
Closes dotnet/roslyn#58790

### CODE REVIEW

File: proposals/csharp-10.0/parameterless-struct-constructors.md [hunk 1]
```diff
@@ -63,6 +63,24 @@ Constructors cannot be `partial`.
...
 > When an instance constructor has no constructor initializer, **or when the constructor initializer `this()` represents the default parameterless constructor**, ... that constructor implicitly performs the initializations specified by the _variable_initializers_ of the instance fields ... . This corresponds to a sequence of assignments that are executed immediately upon entry to the constructor ... . The variable initializers are executed in the textual order in which they appear in the ... declaration.
 
+We also add the following language to [§15.4.9](https://github.com/dotnet/csharpstandard/blob/draft-v6/standard/structs.md#1549-constructors):
+
+> If the struct includes field initializers, then any constructors within the struct which have a `this()` constructor initializer referring to the default parameterless constructor are treated as though they do not have any constructor initializer.
+
+This means we get an error in a scenario like the following:
```
Comment (cston) [inline proposals/csharp-10.0/parameterless-struct-constructors.md:70]: Minor: Perhaps "This means an error is reported for the following:"

**Review by cston — APPROVED** [review]

File: proposals/csharp-10.0/parameterless-struct-constructors.md (hunk 1 above)
Comment (RikkiGibson) [reply proposals/csharp-10.0/parameterless-struct-constructors.md:70]:
```suggestion
This means an error is reported for the following:
```

**Comment by RikkiGibson** [comment]
It looks like we want to change this behavior in 17.2 so that referencing the default parameterless constructor with `: this()` means that we first emit the 'initobj', then the field initializers.

```cs
struct S
{
    int x = 1;
    int y;
    public S(bool unused) : this() { }
}

// compiles to (roughly)
struct S
{
    int x;
    int y;
    public S(bool unused)
    {
        this = default;
        x = 1;
    }
}
```

**Comment by jcouv** [comment]
Should we close this?

**Comment by RikkiGibson** [comment]
Sure. We can open a new PR which specifies the behavior we really want.

**Pull request closed without merging**